OpenRA.Mods.Common/HitShapes/Capsule.cs

```diff
@@ -0,0 +1,75 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Drawing;
+using OpenRA.Graphics;
+
+namespace OpenRA.Mods.Common.HitShapes
+{
+	public class Capsule : IHitShape
+	{
+		public WDist OuterRadius { get { return outerRadius; } }
+		readonly WDist outerRadius;
+
+		readonly int2 v;
+		readonly int2 w;
+		readonly int radius;
+
+		public Capsule(int2 v, int2 w, int radius)
+		{
+			this.v = v;
+			this.w = w;
+			this.radius = radius;
+
+			outerRadius = new WDist(radius + Math.Max(v.Length, w.Length));
+		}
+
+		public WDist DistanceFromEdge(WVec pv)
+		{
+			var p = new int2(pv.X, pv.Y);
+			var vec = w - v;
```

Comment:
`w` and `v` are both constant, so you can calculate `vec` once in the constructor.